src/client/service-worker/events/utility.js

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import {
+  match, put, remove, getMetaData, setMetaData, createCacheName,
+} from '@americanexpress/one-service-worker';
+
+const localeRegExp = /(([a-z]{2,3})(?:-)?([a-zA-Z]{1,})?)\/[^/]*\.json/;
+const legacyRegExp = /legacy\.browser|\/legacy\//;
+
+export function createResourceMetaData(event, resourceInfo, revision) {
+  const [name, baseUrl] = resourceInfo;
+  const [version] = baseUrl.replace(/\/$/, '').split('/').reverse();
+  const { request } = event;
+
+  let type = name === 'app' ? 'one-app' : 'modules';
+
+  let path;
+  let locale;
+  if (localeRegExp.test(request.url)) {
+    type = 'lang-packs';
+    [path, locale] = request.url.match(localeRegExp);
+  }
+
+  let bundle = 'browser';
+  if (legacyRegExp.test(request.url)) {
```

Comment:
can you add some tests around this ? looks like it will still fail for the following
* `cdn.com/modules/legacy/demo.browser`
* `cdn.com/modules/legacy.browser` even though there is a `cdn.com/modules/legacy.legacy.browser`

Reply:
For `cdn.com/modules/legacy/demo.browser`, we need to carry a certain convention for a baseline and we can rely on the bundler extracts for the module file structure (eg `cdn.com/modules/demo/build/1.0.0/demo.browser.js`). If a folder is placed under `/legacy/` we can consider it so.

For `cdn.com/modules/legacy.browser`, I'll update the regexp to `/\.legacy\.browser\./` to expect `*.legacy.browser.*`.

Reply:
we still needs tests to show these scenarios and what is expected.